tests/integration/interface_common_test.py

```diff
@@ -0,0 +1,35 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+import libnmstate
+
+from libnmstate.schema import Interface
+
+
+def test_iface_description_removal(eth1_up):
+    desired_state = eth1_up
+    desired_state[Interface.KEY][0][Interface.DESCRIPTION] = 'bar'
+    libnmstate.apply(desired_state)
+    eth1 = libnmstate.show()
```

Comment:
Unfortunately this is incorrect. A plain ‘show’ will return the full state with all interfaces, not only for eth1.
This means you cannot know which interface exists at index 0.
Also, it is unlikely that the value of the key in question has the same identity as ‘bar’. You need to use equality and not identity.

Reply:
Oh sorry, I knew it is wrong. I just pushed it for my personal use but the PR got updated too. Tomorrow morning I am pushing the fixed version. Thanks!